IPython/core/completer.py

```diff
@@ -117,9 +117,14 @@ def has_open_quotes(s):
 
 def protect_filename(s):
     """Escape a string to protect certain characters."""
+    if set(s) & set(PROTECTABLES):
+        if sys.platform == "win32":
+            return '"' + s + '"'
+        else:
+            return "".join("\\" + c if c in PROTECTABLES else c for c in s)
```

Comment:
I find this a bit hard to read - can we parenthesise the ternary expression (i.e. so the close-paren is just before `for`).